### PR TITLE
Use centos stream 9 as base image

### PR DESCRIPTION
**Pull request by elfosardo**

### CODE REVIEW

**Comment by elfosardo** [comment]
the image builds just with the base image change, we need to test the workflow

**Comment by dtantsur** [comment]
We don't have any upstream tests for this container, le sigh.

**Comment by elfosardo** [comment]
> We don't have any upstream tests for this container, le sigh.

the only test I did is a build test, the image builds just fine
of course not having integration test is not ideal :/

**Comment by dtantsur** [comment]
Could you add the tzdata workaround here as well? This repo is affected.

**Comment by dtantsur** [comment]
With the workaround IPA at least tries to start.

File: prepare-image.sh [hunk 1]
```diff
@@ -28,6 +28,10 @@ rpm -q subscription-manager && \
 # Pbr pulls in Git (30+ MiB), but actually only uses it in development context.
 rpm -q git-core && rpm -e --nodeps git-core || true
 
+# NOTE(elfosardo): we need to reinstall tzdata as the base CS9 container removes
+# its content, for more info see https://bugzilla.redhat.com/show_bug.cgi?id=2052861
+dnf reinstall -y tzdata
```
Comment (dtantsur) [inline prepare-image.sh:33]: Please move this before the `rpm` commands above because they invalidate the dependencies

Comment (elfosardo) [reply prepare-image.sh:33]: oh snap! yeah...

**Review by iurygregory — APPROVED** [review]
/lgtm

**Pull request merged**